src/main/java/vg/civcraft/mc/citadel/activity/ActivityListener.java

```diff
@@ -0,0 +1,74 @@
+package vg.civcraft.mc.citadel.activity;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerJoinEvent;
+import org.bukkit.event.player.PlayerMoveEvent;
+import vg.civcraft.mc.citadel.Citadel;
+import vg.civcraft.mc.citadel.CitadelPermissionHandler;
+import vg.civcraft.mc.namelayer.GroupManager;
+import vg.civcraft.mc.namelayer.NameAPI;
+import vg.civcraft.mc.namelayer.group.Group;
+
+public class ActivityListener implements Listener {
+
+	private final ActivityMap map;
+	private final int resolution;
+
+	public ActivityListener(ActivityMap map) {
+		this.map = map;
+		this.resolution = Citadel.getInstance().getConfigManager().getActivityMapResolution();
+	}
+
+	@EventHandler
+	public void on(PlayerMoveEvent event) {
+		Location from = event.getFrom();
+		Location to = event.getTo();
+		//noinspection ConstantConditions
+		if (to == null || from == null || !map.isEnabled(to.getWorld())) {
+			return;
+		}
+
+		int fsx = from.getBlockX() / resolution;
```

Comment:
Need to round down, otherwise errors around 0, because:

-200 / 512 = 0
200 / 512 = 0